### PR TITLE
Only allow unique verification keys to register

### LINKED ISSUE #47: ensure global uniqueness of user-provided verification keys
If I understand correctly, we blindly register the user-provided verification key and then allow player-player identification using it. It's fine but if we do not ensure uniqueness, there must be some disruption vectors/bugs waiting. Some of the cases are handled by other tests but it should be better to be certain about it.

For example, right from the beginning, a user registering with a duplicate vk will simply overwrite the existing user. Then we have a mixed user state with one existing by connection but not by vk.

### PR DESCRIPTION
**Pull request by emergent-reasons**
Fixes #47 

I took a shot at solving this one, assuming that the issue I filed is valid in the first place.

### CODE REVIEW

**Comment by emergent-reasons** [comment]
This is based on my `fail-fast` branch from #49 because it needs to have accurate and fast detection of a closed connection.

File: server/register.go [hunk 1]
```diff
@@ -2,22 +2,31 @@ package server
...
-			if p.FromKey.String() != "" && registration != nil {
+			verificationKey := p.FromKey.String()
+			player = pi.tracker.playerByVerificationKey(verificationKey)
+			if player != nil {
+				return fmt.Errorf("server already has a player "+
+					"with verification key %s", verificationKey)
+			}
+
+			if verificationKey != "" && registration != nil {
 				player := playerData{
```
Comment (zquestz) [inline server/register.go]: NIT: shouldn't this just be `player =` since it is defined above?

**Comment by zquestz** [comment]
Has minor conflicts from the previous merge. Can merge this once they are fixed.

File: server/register.go (hunk 1 above)
Comment (emergent-reasons) [reply server/register.go]: Yes definitely. Thank you. 0be99d3

**Pull request merged**